### DISCUSSION TITLE
Themes seem to sometimes not work through WSL

### DISCUSSION THREAD

**Discussion by jquesada2016** [helix-editor/helix#7163] [General] [answered]
### Summary

When running Helix within WSL, through a terminal emulator, such as the new Windows terminal emulator, or Alacritty, then themes seem to not be applied, and instead I get only black-and-white styling:

![image](https://github.com/helix-editor/helix/assets/54370171/4522b4f8-5147-4e70-9644-ba72b5166ec7)

However, when I access helix through WSL, from, for example an integrated terminal within VS Code, themes work fine!

![image](https://github.com/helix-editor/helix/assets/54370171/a5b80fb9-cfcd-4b5d-b111-9320db3ae993)

Not sure what's going on here. The runtime files, and themes definitely seem to be there, since VS Code's terminal is able to pick up the colors.

### Reproduction Steps

 

I tried installing from package manager, as well as building from lates master, then switched to latest 23.05. I also tried using both `HELIX_RUNTIME` environment variables, as well as symlinking the runtime files into `~/.config/helix/runtime`. Nothing worked, besides running Helix within WSL inside VS Code

I expected this to happen:

Themes to work, regardless of the terminal emulator I used.

Instead, this happened:

Theme colors only seemed to be applied when attaching through VS Code's integrated terminal.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-05-27T19:31:59.413 helix_view::editor [ERROR] editor error: Unsupported theme: theme requires true color support
```

</details>

### Platform

Ubuntu through WSL on Windows 11

### Terminal Emulator

Alacritty v0.11.0 and Windows Terminal version 1.16.10262.0

### Helix Version

helix 23.05

**Comment by jquesada2016** [comment]
Looking at the log attached above, I noticed it says true color support is required, which is strange, because both Alacritty and Windows Terminal support true color...so not sure how to trubleshoot this.

**Comment by kirawi** [comment, marked as answer]
It's not picked up when running WSL. You can override detection in your config: https://docs.helix-editor.com/configuration.html

**Reply by gabydd** [reply]
specifically with:

```toml
[editor]
true-color = true
```

**Reply by jquesada2016** [reply]
Thank you very much! Might not be a bad idea to include this as a note for WSL users in the installation docs.

**Reply by nowhereman1999** [reply]
Thank you very much! It's a good idea to include a note for WSL users in the installation instructions. I searched extensively in the documentation and online but couldn't find a solution. Fortunately I found the same issue in the GitHub issues. We should definitely add a special note in the docs to warn WSL2 users about this

**Reply by nowhereman1999** [reply]
sorry to bother here again,but I found a very weird thing,the theme I set display correctly when I just type 'hx' to open it as shown below:
![image](https://github.com/helix-editor/helix/assets/53631206/dc872d8f-e1d8-424f-bb30-3675789dd82f)
but it soon became only black-and-white styling when I type 'sudo hx':
![image](https://github.com/helix-editor/helix/assets/53631206/bb90c9cd-86e1-45a5-bbd7-b4f75a9bcb09)
So why it exhibits two different behaviors just because I put sudo before the hx command?hope you guys can give me some advice！

**Reply by gabydd** [reply]
when you use ``sudo`` you are running the program as the root user which won't use your user config(so it doesn't use the theme you set). You can either make a config for root or see sudoedit: https://github.com/helix-editor/helix/discussions/4251

**Reply by nowhereman1999** [reply]
ok,I have checked the issue you mentioned and according to the explanation in that issue, do I just need to set Helix as the default editor and then use `sudoedit` instead of  `sudo hx` or create a separate config file for root, in order to use `sudo hx` normally?"

**Reply by nowhereman1999** [reply]
By the way,I noticed that after opening `helix` in WSL2, there's always a certain gap around the surroundings and the terminal, as indicated by the red rectangle in the image below. Is this also caused by the settings in the configuration?
![image](https://github.com/helix-editor/helix/assets/53631206/700eaf72-49dd-4855-bde8-0a0fa930ca25)

**Reply by KamWithK** [reply]
I also see the border and was wondering that too

**Reply by kirawi** [reply]
The border is done by the terminal, and resizing the terminal window itself would fix it. IIRC, other editors send a control code to change the terminal background so it's not noticeable, but this isn't implemented in Helix yet.

**Reply by KamWithK** [reply]
Decreasing the font size down to 12 seems to decrease the border by quite a lot
I don't think resizing the window does anything for mine though

**Reply by KamWithK** [reply]
I tested it out a little more and it seems like resizing the window manually (dragging it out) does seem to fix the problem, *however*, snapping it to the grid (like left half of the screen) will cause the problem to come back
Not sure if this happens with Windows by default or if it's just powertoys which it happens to though (I can't tell since I have powertoys installed and set up on both my computers)
Is there a command or something that can be run on helix startup to fix it instead?

**Reply by nowhereman1999** [reply]
Hi,the gap is still exists when I resizing the window manually like the pictures shown below:
![image](https://github.com/helix-editor/helix/assets/53631206/68762f61-923e-4abf-b33b-f6dfaf071083)
![image](https://github.com/helix-editor/helix/assets/53631206/afad4cd9-5b78-4018-a65f-ae5c9e6f11a3)
I also install powertoys in my computer,so is this the reason causing the problem?By the way,there is another problem when I try to open file using sudo as I mentioned here,did you also meet the problem too?

**Reply by kirawi** [reply]
You might have more luck checking out Windows Terminal resources. This isn't a Helix issue

**Reply by KamWithK** [reply]
I haven't tried running as sudo yet
Normally good practice not to anyways aha

**Reply by nowhereman1999** [reply]
oh sorry,I didn't put the [link ](https://github.com/helix-editor/helix/discussions/4251#discussioncomment-6647977 ) that I mentioned before,actually there does have the requirement for using `sudo hx file` so that you can have rights to write to it,but the problem
 is when I use `sudo hx` to open file,the config loaded in helix is no longer mine as you can see the theme is back to the default  black-and-white styling:
 ![image](https://github.com/helix-editor/helix/assets/53631206/61f9b0b7-f9d1-4dde-97d0-2b423466cc5e)
when the theme loaded normally as I just using `hx` to open the file:
![image](https://github.com/helix-editor/helix/assets/53631206/96626626-7c14-4a5f-a088-cfabb80917aa)

**Reply by dawkrish** [reply]
there is gap 
i can see wsl terminal at boundaries and hx in the center
how to remove that

**Reply by BrianShTsoi** [reply]
You can change it with Windows Teminal Settings. 
https://stackoverflow.com/questions/62578458/vim-nvim-not-filling-windows-terminal

**Comment by knightpp** [comment]
The selected answer did not help me, but here's my solution.

To debug open hx and type `:theme <tab>` you will probably see
![image](https://github.com/helix-editor/helix/assets/28928944/271a894b-529f-47b7-a8bf-cd3854718450)
To fix that, I set `TERMCOLOR` env var to `truecolor` ([source](https://askubuntu.com/a/1329392))

**Reply by kirawi** [reply]
It's very likely you did not place the config in the correct place, as at the very least Helix could not read the true color setting. One common mistake people make with WSL is keeping the config on the Windows side while Helix itself is installed in the Linux side.

**Reply by knightpp** [reply]
Oh, I see the issue now, I did not understand your answer. Thanks.

I needed to set true-color in `~/.config/helix/config.toml`
```toml
[editor]
true-color = true
```

**Reply by senthilnow** [reply]
[editor]
true-color = true

This worked :-) +1

**Reply by ByteJuggler** [reply]
Can also set in the env e.g. execute in shell `export COLORTERM=truecolor` or add this line to say `~/.bashrc`, then you don't need to override in `config.toml`.
